File: AN_2/SEMESTRUL_III/MAP/LABS/LABORATOR 6/SAPT10GUI - saptamana 10/src/main/java/com/repo/Repository.java
package repo;

import java.util.List;

public interface Repository<ID,entityType> {
    void adauga(entityType entity);
    void sterge(entityType entity);
    entityType cautaId(String id);
    List<entityType> getAll();
    void update(entityType entitate,entityType nouaEntitate);
}

File: AN_2/SEMESTRUL_IV/MPP/LABORATOR/CSHARP_TEMA/tema_sapt3/Persistance/Repository.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.Design;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace persistance
{
    public interface Repository<ID, entityType> where entityType: Entitate<ID>
    {
        void adauga(entityType entity);
        void sterge(entityType entity);
        entityType cautaId(ID id);
        List<entityType> getAll();
        void update(entityType entitate, entityType nouaEntitate);
    }
}
